### PR TITLE
Crash when auto incremented column updated in partitioned table

### PR DESCRIPTION
**Pull request by voodam**
https://jira.percona.com/browse/PS-7856

Use dbug_tmp_use_all_columns/dbug_tmp_restore_column_map as described in 74cc73d as a workaround, because autoincremented column is not used in the query.

This is new PR instead of https://github.com/percona/percona-server/pull/4567 to comply GCA process.

### CODE REVIEW

**Review by satya-bodapati — CHANGES_REQUESTED** [review]
1. please fix commit message and github description. I think the write_set part is not relevant anymore
2. This doesn't look a GCA commit. If this is fixed in 5.7, we should see two commits for 8.0 branch

**Comment by voodam** [comment]
> 1. please fix commit message and github description. I think the write_set part is not relevant anymore

Done, my bad.

> 2. This doesn't look a GCA commit. If this is fixed in 5.7, we should see two commits for 8.0 branch

You mean second commit with resolving conflicts? Yes, but AFAIU I first need to merge PR for 5.7: https://github.com/percona/percona-server/pull/4589, otherwise this 5.7 PR will have wrong commit. Or is this wrong undestanding of GCA process?

UPD. I probably do things in the wrong order: I need to create 8.0 PR after merging 5.7, not at the same time, right? Now it's need to wait while 5.7 PR will be merged.

File: mysql-test/suite/innodb/t/partition_autoinc.test [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#################################
```
Comment (kamil-holubicki) [inline mysql-test/suite/innodb/t/partition_autoinc.test:3]: `--source include/have_innodb.inc`

Comment (voodam) [reply mysql-test/suite/innodb/t/partition_autoinc.test:3]: Done (for 5.7)

**Comment by voodam** [comment]
I need to close PR again because of missunderstanding GCA, sorry for the mess. Now all should be OK.
Here is the new one: https://github.com/percona/percona-server/pull/4603

**Pull request closed without merging**